### PR TITLE
[Nemo-781] Support SparkSQL

### LINKED ISSUE #781: Fully support SparkSQL
Sub-issue of #776 
It should support the Dataset / Dataframe related methods for SparkSQL
See package edu.snu.nemo.compiler.frontend.spark.sql

### PR DESCRIPTION
**Pull request by wonook**
This PR:

- Enables support for various SparkSQL-related methods in Dataset, DataFrameReader, and SparkSession classes.
- minor: fixes wrong licenses

resolves #781

### CODE REVIEW

File: compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/sql/NemoSparkSQL.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.nemo.compiler.frontend.spark.sql;
+
+/**
+ * Common interface for Nemo Spark classes.
+ */
+public interface NemoSparkSQL {
```
Comment (sanha) [inline compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/sql/NemoSparkSQL.java]: Please rename this interface. IT is not that informative.

Comment (sanha) [reply compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/sql/NemoSparkSQL.java]: Please update the comment also.

Comment (wonook) [reply compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/sql/NemoSparkSQL.java]: I've addressed your comment 😄

Comment (wonook) [reply compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/sql/NemoSparkSQL.java]: @sanha

**Pull request merged**